Modules/_sqlite/connection.c

```diff
@@ -432,48 +436,75 @@ free_callback_contexts(pysqlite_Connection *self)
 static void
 remove_callbacks(sqlite3 *db)
 {
+    if (db == NULL) {
```

Comment:
I would prefer to check that in the caller. Maybe add an assertion if you want.

Reply:
I tried both variants; I have no strong preference, so I'll change it to be the responsibility of the caller.

Reply:
I changed the responsibility and added an assert.